app/src/main/java/com/arctouch/codechallenge/util/NetworkObserver.java

```diff
@@ -0,0 +1,17 @@
+package com.arctouch.codechallenge.util;
+
+import android.app.Application;
+import android.content.Context;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+
+public class NetworkObserver {
+
+    public boolean isNetworkAvailable(Application context) {
+        ConnectivityManager connectivityManager
+                = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
+        assert connectivityManager != null;
```

Comment:
Why did you use `assert` here? What does it do?

Reply:
Assertions helps checking for errors in your code logic, but shouldn't be on a release code, just debugging. It should be replaced with a conditional or a try catch statement. The main problem here is that the assert will throw AssertionError which extends Error not Exception, crashing the app.